### PR TITLE
[orchagent] Set ABRT signal in STATE_DB during a SAI failure

### PR DESCRIPTION
**Pull request by vivekrnv**
Signed-off-by: Vivek Reddy Karri <vkarri@nvidia.com>

**What I did**

- During a SAI programming failure, made orchagent set the ORCH_ABRT_STATUS flag in STATE_DB. Also, delete the flag once the orchagent restarts

**Why I did it**

**How I verified it**

Simulate a SAI failure:
```
81396-Nov 21 18:14:59.131240 mtbc-sonic-01-2410 ERR swss#orchagent: :- create: create status: SAI_STATUS_INVALID_ATTRIBUTE_MAX
81397-Nov 21 18:14:59.131240 mtbc-sonic-01-2410 ERR swss#orchagent: :- sflowCreateSession: Failed to create sample packet session with rate 512
81398:Nov 21 18:14:59.131240 mtbc-sonic-01-2410 ERR swss#orchagent: :- handleCreate: Encountered failure in create operation, exiting orchagent, SAI API: SAI_API_SAMPLEPACKET, status: SAI_STATUS_INVALID_ATTRIBUTE_MAX
```

Check if STATE_DB is updated and cleared once the orchagent is restarted
```
root@mtbc-sonic-01-2410:/home/admin# sonic-db-cli STATE_DB GET ORCH_ABRT_STATUS
1
root@mtbc-sonic-01-2410:/home/admin#

root@mtbc-sonic-01-2410:/home/admin# sonic-db-cli STATE_DB GET ORCH_ABRT_STATUS

root@mtbc-sonic-01-2410:/home/admin#
```

**Details if related**

### CODE REVIEW

File: orchagent/main.cpp [hunk 1]
```diff
@@ -486,6 +486,9 @@ int main(int argc, char **argv)
     DBConnector config_db("CONFIG_DB", 0);
     DBConnector state_db("STATE_DB", 0);
 
+    /* Clears the ORCH_ABORT_STATUS flag in STATE_DB */
+    state_db.del(ORCH_ABRT);
```
Comment (prsunny) [inline orchagent/main.cpp:490]: suggest have flush table at common place - https://github.com/sonic-net/sonic-buildimage/blob/master/files/scripts/swss.sh#L177

File: orchagent/orch.cpp [hunk 2]
```diff
@@ -23,6 +23,15 @@ extern ofstream gRecordOfs;
 extern bool gLogRotate;
 extern string gRecordFile;
 
+void notifyAbort(){
```
Comment (prsunny) [inline orchagent/orch.cpp:26]: API name is not aligned to the functionality. This is doing both notify and abort. API gives the indication that it is only notification. Please move abort to original code and use it only to notify or change the name to say `notifyAndAbort`

File: orchagent/main.cpp (hunk 1 above)
Comment (vivekrnv) [reply orchagent/main.cpp:490]: That can be done, but clearing it here gives more buffer to the processes dependent on this flag.

File: orchagent/orch.cpp (hunk 2 above)
Comment (vivekrnv) [reply orchagent/orch.cpp]: Handled

**Review by dgsudharsan — APPROVED** [review]

**Pull request closed without merging**